packages/@react-spectrum/s2/src/Tabs.tsx

```diff
@@ -175,63 +169,50 @@ const tablist = style({
 });
 
 export function TabList<T extends object>(props: TabListProps<T>) {
-  let {density, isDisabled, disabledKeys, orientation} = useSlottedContext(TabsContext) ?? {};
+  let {density, isDisabled, disabledKeys, orientation} = useContext(InternalTabsContext) ?? {};
+  let {showItems} = useContext(CollapseContext) ?? {};
   let state = useContext(TabListStateContext);
   let [selectedTab, setSelectedTab] = useState<HTMLElement | undefined>(undefined);
   let tablistRef = useRef<HTMLDivElement>(null);
 
   useLayoutEffect(() => {
-    if (tablistRef?.current) {
+    if (tablistRef?.current && showItems) {
       let tab: HTMLElement | null = tablistRef.current.querySelector('[role=tab][data-selected=true]');
 
       if (tab != null) {
         setSelectedTab(tab);
       }
+    } else if (tablistRef?.current) {
+      let picker: HTMLElement | null = tablistRef.current.querySelector('button');
```

Comment:
I forget if aria disallows it, but is the picker guaranteed to be the only button element in the tablist? I don't see `children presentation` for the tablist in the spec

Reply:
yeah, the role tablist doesn't allow for anything other than tabs, I may need to switch the role when we collapse now that you bring it up...

Reply:
@majornista thoughts?